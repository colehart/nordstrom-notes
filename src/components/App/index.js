import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Header from '../../containers/Header'
import NoteForm from '../../containers/NoteForm'
import NotesList from '../../containers/NotesList'
import './App.css';

export const App = () => {
  const noMatch = ({ location }) => (
    <div className='a-404'>
      <h2>404 - Path not found.</h2>
      <p>No routes match {location.pathname}.</p>
      <p>Please click the the 'Take Note' or 'Notable Notes' buttons or your browser's forward and back buttons to navigate to your desired destination. The Nordstrom Notes logo will return you to the Take Note page.</p>
    </div>
  )

  return (
    <div className="App">
      <Header />
      <Switch>
        <Route
          exact path='/'
          component={ NoteForm }
        />
        <Route
          exact path='/notes'
          component={ NotesList }
        />
        <Route
          component={ noMatch }
        />
      </Switch>
    </div>
  );
}

