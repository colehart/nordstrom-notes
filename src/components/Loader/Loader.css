.Loader {
  grid-row: 2 / 3;
  width: 100%;
  height: 0;
  padding-bottom: 67%;
  position: relative;
}

.l-giphy-embed {
  width: 100%;
  height: 100%;
  position: absolute;
}