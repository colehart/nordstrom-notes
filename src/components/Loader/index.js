import React from 'react'
import './Loader.css'

export const Loader = () => {
  return (
    <div className='Loader'>
      <iframe
        className="l-giphy-embed"
        src="https://giphy.com/embed/xkmQfH1TB0dLW"
        frameBorder="0"
        title="typewriter"
        allowFullScreen
      >
      </iframe>
    </div>
  )
}