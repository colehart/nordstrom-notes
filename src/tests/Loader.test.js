import React from 'react'
import { shallow } from 'enzyme'
import { Loader } from '../components/Loader'

describe('Loader', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Loader />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})