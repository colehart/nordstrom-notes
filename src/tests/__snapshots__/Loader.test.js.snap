// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should match the snapshot 1`] = `
<div
  className="Loader"
>
  <iframe
    allowFullScreen={true}
    className="l-giphy-embed"
    frameBorder="0"
    src="https://giphy.com/embed/xkmQfH1TB0dLW"
    title="typewriter"
  />
</div>
`;
